core/src/main/scala/com/rallyhealth/vapors/core/math/Arithmetic.scala

```diff
@@ -0,0 +1,113 @@
+package com.rallyhealth.vapors.core.math
+
+import cats.{Apply, Semigroupal}
+import com.rallyhealth.vapors.core.math.Arithmetic.{Addition, Subtraction}
+
+object Arithmetic extends AdditionTuples with SubtractionTuples {
+
+  final object Implicits {
+
+    implicit class Ops[A](private val lhs: A) extends AnyVal {
+
+      def +(rhs: A)(implicit A: Addition[A]): A = A.add(lhs, rhs)
+
+      def -(rhs: A)(implicit A: Subtraction[A]): A = A.subtract(lhs, rhs)
+    }
+  }
+
+  def zero[A](implicit A: Zero[A]): A = A.zero
+
+  trait Zero[A] {
+    def zero: A
+  }
+
+  object Zero {
+    implicit def fromCatsZero[A](implicit A: alleycats.Zero[A]): Zero[A] = new Zero[A] {
+      override def zero: A = A.zero
+    }
+  }
+
+  trait Addition[A] {
+
+    def add(
+      lhs: A,
+      rhs: A,
+    ): A
+  }
+
+  trait Subtraction[A] {
+
+    def subtract(
+      lhs: A,
+      rhs: A,
+    ): A
+  }
+
+  type Linear[A] = Addition[A] with Subtraction[A]
+
+  trait Negative[A] {
+
+    def negative(value: A): A
+  }
+
+  type Numerical[A] = Linear[A] with Negative[A] with Zero[A]
+
+  implicit def numeric[A](implicit A: Numeric[A]): Numerical[A] = new FromNumeric[A]
+
+  private final class FromNumeric[A : Numeric] extends Addition[A] with Subtraction[A] with Negative[A] with Zero[A] {
+    import Numeric.Implicits._
+
+    override def zero: A = Numeric[A].zero
+
+    override def add(
+      lhs: A,
+      rhs: A,
+    ): A = lhs + rhs
+
+    override def subtract(
+      lhs: A,
+      rhs: A,
+    ): A = lhs - rhs
+
+    override def negative(value: A): A = -value
+  }
+}
+
+trait AdditionTuples {
+
+  import Arithmetic.Implicits._
+
+  implicit def addTuple2[A : Addition, B : Addition]: Addition[(A, B)] = {
+    case ((a1, b1), (a2, b2)) => (a1 + a2, b1 + b2)
+  }
+
+  implicit def addTuple3[A : Addition, B : Addition, C : Addition]: Addition[(A, B, C)] = {
+    case ((a1, b1, c1), (a2, b2, c2)) => (a1 + a2, b1 + b2, c1 + c2)
+  }
+
+  implicit def addTuple4[A : Addition, B : Addition, C : Addition, D : Addition]: Addition[(A, B, C, D)] = {
+    case ((a1, b1, c1, d1), (a2, b2, c2, d2)) => (a1 + a2, b1 + b2, c1 + c2, d1 + d2)
```

Comment:
I don't think this is necessary. Adding tuples is appealing, but it is not very likely to be useful. Especially since you can nest expressions as needed to define addition on the individual values without packing them into a tuple first. I'm gonna remove this stuff for now.

Reply:
Deleted this code.